src/uu/stat/src/stat.rs

```diff
@@ -293,6 +318,93 @@ fn print_str(s: &str, flags: &Flags, width: usize, precision: Option<usize>) {
     pad_and_print(s, flags.left, width, Padding::Space);
 }
 
+fn quote_file_name(file_name: &str, quoting_style: &QuotingStyle) -> String {
+    match quoting_style {
+        QuotingStyle::Locale | QuotingStyle::Shell => {
+            let escaped = file_name.replace('\'', r"\'");
+            format!("'{}'", escaped)
+        }
+        QuotingStyle::Default => format!("\"{}\"", file_name),
+        QuotingStyle::Quote => file_name.to_string(),
+    }
+}
+
+fn get_quoted_file_name(
+    display_name: &str,
+    file: &OsString,
+    file_type: &FileType,
+    from_user: bool,
+) -> Result<String, i32> {
+    let quoting_style = env::var("QUOTING_STYLE")
+        .ok()
+        .and_then(|style| style.parse().ok())
+        .unwrap_or(QuotingStyle::Default);
+
+    if file_type.is_symlink() {
+        let quoted_display_name = quote_file_name(display_name, &quoting_style);
+        match fs::read_link(file) {
+            Ok(dst) => {
+                let quoted_dst = quote_file_name(&dst.to_string_lossy(), &quoting_style);
+                Ok(format!("{quoted_display_name} -> {quoted_dst}"))
+            }
+            Err(e) => {
+                show_error!("{e}");
+                Err(1)
+            }
+        }
+    } else {
+        let style = if from_user {
+            quoting_style
+        } else {
+            QuotingStyle::Quote
+        };
+        Ok(quote_file_name(display_name, &style))
+    }
+}
+
+fn process_token_fs(t: &Token, meta: StatFs, display_name: &str) {
```

Comment:
What does `fs` in the function name mean in this context?

Reply:
filesystem, but i will rename it :)